selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsProjectsTest.java

```diff
@@ -0,0 +1,199 @@
+  @Inject private TestWorkspaceServiceClient testWorkspaceServiceClient;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+  @Inject private DefaultTestUser defaultTestUser;
+  @Inject private SeleniumWebDriverHelper seleniumWebDriverHelper;
+  @Inject private ProjectExplorer projectExplorer;
+
+  @BeforeClass
+  public void setup() throws Exception {
+
+    Path projectPath =
+        Paths.get(getClass().getResource("/projects/default-spring-project").toURI());
+    testProjectServiceClient.importProject(
+        testWorkspace.getId(), projectPath, PROJECT_NAME, MAVEN_SPRING);
+
+    testProjectServiceClient.importProject(
+        testWorkspace.getId(), projectPath, PROJECT_FOR_SEARCHING_NAME, MAVEN_SPRING);
+
+    testProjectServiceClient.importProject(
+        checkingCheckboxesWorkspace.getId(),
+        projectPath,
+        CHECKING_CHECKBOXES_PROJECT_NAME,
+        MAVEN_SPRING);
+
+    testProjectServiceClient.importProject(
+        checkingCheckboxesWorkspace.getId(),
+        projectPath,
+        CHECKING_CHECKBOXES_NEVEST_PROJECT_NAME,
+        MAVEN_SPRING);
+
+    testWorkspaceServiceClient.stop(
+        checkingCheckboxesWorkspace.getName(), defaultTestUser.getName());
+
+    dashboard.open();
+    dashboard.waitDashboardToolbarTitle();
+    dashboard.selectWorkspacesItemOnDashboard();
+    workspaces.waitToolbarTitleName();
+    workspaces.selectWorkspaceItemName(testWorkspace.getName());
+    workspaceDetails.waitToolbarTitleName(testWorkspace.getName());
+    workspaceDetails.selectTabInWorkspaceMenu(PROJECTS_TAB_NAME);
+  }
+
+  @Test
+  public void mainElementsShouldBePresent() throws Exception {
+    openWoprkspaceDetailsProjectsPage(testWorkspace.getName());
+
+    addSpringSampleToWorkspace();
+
+    checkProjectAppearanceAndButtonsState();
+
+    workspaceProjects.clickOnBottomButton(WorkspaceProjects.BottomButton.CANCEL_BUTTON);
+    workspaceProjects.waitProjectIsNotPresent(SPRING_SAMPLE_NAME);
+
+    addSpringSampleToWorkspace();
+
+    checkProjectAppearanceAndButtonsState();
+
+    workspaceProjects.clickOnBottomButton(SAVE_BUTTON);
+
+    workspaceProjects.waitNotification(EXPECTED_SUCCESS_NOTIFICATION);
+    workspaceProjects.waitProjectIsPresent(SPRING_SAMPLE_NAME);
+  }
+
+  @Test
+  public void checkSearchField() throws Exception {
+    final String textForSearching = "sea";
+
+    openWoprkspaceDetailsProjectsPage(testWorkspace.getName());
+
+    workspaceProjects.waitSearchField();
+    workspaceProjects.typeToSearchField(textForSearching);
+    workspaceProjects.waitProjectIsNotPresent(PROJECT_NAME);
+    workspaceProjects.waitProjectIsPresent(PROJECT_FOR_SEARCHING_NAME);
+
+    workspaceProjects.typeToSearchField("");
+    workspaceProjects.waitProjectIsPresent(PROJECT_NAME);
+    workspaceProjects.waitProjectIsPresent(PROJECT_FOR_SEARCHING_NAME);
+  }
+
+  @Test(priority = 1)
+  public void checkOfCheckboxes() throws Exception {
```

Comment:
it's not quite clear what kind of check this test is about. It could be expressed in method name, or explained in comment above the method definition, or by commenting each steps inside the test.